test/infrastructure/docker/e2e/docker_suite_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/generators"
+
+	corev1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha2"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha2"
+)
+
+func TestDocker(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Docker Suite")
+}
+
+var mgmt *CAPDCluster
+var ctx = context.Background()
+
+var _ = BeforeSuite(func() {
+	managerImage := os.Getenv("MANAGER_IMAGE")
+	if managerImage == "" {
+		managerImage = "gcr.io/k8s-staging-capi-docker/capd-manager-amd64:dev"
+	}
+	By("setting up in BeforeSuite")
+	var err error
+
+	// Set up the provider component generators based on master
+	core := &generators.ClusterAPI{GitRef: "master"}
+
+	// set up capd components based on current files
+	infra := &provider{}
+
+	// set up cert manager
+	cm := &generators.CertManager{ReleaseVersion: "v0.11.0"}
+
+	scheme := runtime.NewScheme()
+	Expect(v1.AddToScheme(scheme)).To(Succeed())
+	Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	Expect(bootstrapv1.AddToScheme(scheme)).To(Succeed())
+	Expect(infrav1.AddToScheme(scheme)).To(Succeed())
+
+	// Create the management cluster
+	mgmt, err = NewClusterForCAPD(ctx, "mgmt", scheme)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(mgmt).NotTo(BeNil())
+
+	Expect(mgmt.LoadImage(ctx, managerImage)).NotTo(HaveOccurred())
```

Comment:
Would it make sense to add a slice of manager images to load to the signature for `NewClusterForCAPD` rather than having a separate call here?

Reply:
I'm wondering if it would also make sense to push it up into the kind management cluster helper as well?

Reply:
You're suggesting changing the cluster create signature from

```go
func NewClusterForCAPD(ctx context.Context, name string, scheme *runtime.Scheme) (*CAPDCluster, error)
```

to

```go
func NewClusterForCAPD(ctx context.Context, name string, scheme *runtime.Scheme, images ...string) (*CAPDCluster, error) {
```

is that right?

Reply:
Yes, I believe it would help reduce the amount of boilerplate needed for each consumer of the framework.

Reply:
yeah, it's a good idea especially considering this is required to load the locally built image.